control/cephutils.py

```diff
@@ -234,6 +236,42 @@ def get_image_size(self, pool_name: str, image_name: str) -> int:
 
         return image_size
 
+    def were_image_qos_limits_changed(self, pool_name: str, image_name: str) -> bool:
+        if not self.pool_exists(pool_name):
+            raise rbd.ImageNotFound(f"Pool {pool_name} doesn't exist", errno=errno.ENODEV)
+
+        with rados.Rados(conffile=self.ceph_conf, rados_id=self.rados_id) as cluster:
+            with cluster.open_ioctx(pool_name) as ioctx:
+                try:
+                    with rbd.Image(ioctx, image_name) as img:
+                        attributes = img.config_list()
+                        self.logger.debug(f"Config for image {pool_name}/{image_name}:")
+                        self.logger.debug("==========================================")
+                        for one_img_attr in attributes:
+                            self.logger.debug(f"{one_img_attr}")
+                            try:
+                                if not one_img_attr["name"].startswith(CephUtils.RBD_QOS_PREFIX):
+                                    continue
+                                if not one_img_attr["name"].endswith(CephUtils.RBD_QOS_SUFFIX):
+                                    continue
+                                if one_img_attr["value"] != "0":
+                                    self.logger.warning(f'RBD QOS attribute '
+                                                        f'{one_img_attr["name"]} was changed '
+                                                        f'from 0 to '
+                                                        f'{one_img_attr["value"]}')
+                                    return True
+                            except Exception:
+                                self.logger.exception(f"error parsing {one_img_attr}")
+                except rbd.ImageNotFound:
+                    raise rbd.ImageNotFound(f"Image {pool_name}/{image_name} doesn't exist",
+                                            errno=errno.ENODEV)
+                except Exception as ex:
+                    self.logger.exception(f"Error while trying to get the config of image "
+                                          f"{pool_name}/{image_name}")
+                    raise ex
```

Comment:
I think in python its better to just do it as this:
```suggestion
                    raise
```
as in [python documentation](https://docs.python.org/2/reference/simple_stmts.html#the-raise-statement):
`If no expressions are present, [raise](https://docs.python.org/2/reference/simple_stmts.html#raise) re-raises the last exception that was active in the current scope.`

Reply:
or raise a dedicated exception, for example if the dedicated example is called `RbdException`, so we will have the original exception context
```suggestion
                    raise RbdException(f"{pool_name}/{image_name} errored") from ex
```

Reply:
Originally I used ex in the message. I changed that and forgot to get rid of ex in the other places. Will do that.